app/controllers/webapp/Profile.scala

```diff
@@ -247,31 +247,108 @@ class Profile @Inject() (
+      //case Some(user) => Future.successful(WebAppResult.Ok(request, "profile.requestNVM.success", Nil, "Profile.requestNVM.success", Json.obj("status" -> "in progress")).getResult)
+      case None => Future.successful(WebAppResult.Unauthorized(request, "error.noAuthenticatedUser", Nil, "AuthProvider.Identity.Unauthorized", Map[String, String]()).getResult)
+    }
+  }
+
   def requestNVM = UserAwareAction.async { implicit request =>
     request.identity match {
+      case Some(user) => {
+        user.profiles.headOption match {
+          case Some(profile) => userService.activeNVM(profile).map(status =>
+            WebAppResult.Ok(request, "profile.requestNVM.success", Nil, "Profile.requestNVM.success", Json.obj("status" -> "active")).getResult
+          )
+          case None => Future.successful(WebAppResult.NotFound(request, "profile.requestNVM.notFound", Nil, "Profile.requestNVM.notFound", Map[String, String]()).getResult)
+        }
+      }
       //dummy function. Validation test not implemented
-      case Some(user) => Future.successful(WebAppResult.Ok(request, "profile.requestNVM.success", Nil, "Profile.requestNVM.success", Json.obj("status" -> "in progress")).getResult)
+      //case Some(user) => Future.successful(WebAppResult.Ok(request, "profile.requestNVM.success", Nil, "Profile.requestNVM.success", Json.obj("status" -> "in progress")).getResult)
       case None => Future.successful(WebAppResult.Unauthorized(request, "error.noAuthenticatedUser", Nil, "AuthProvider.Identity.Unauthorized", Map[String, String]()).getResult)
     }
   }
-  
+
   def checkNVM = UserAwareAction.async { implicit request =>
     request.identity match {
       //dummy function. Need Validation from requestNVM for check if a user is NVM
-      case Some(user) => Future.successful(WebAppResult.Ok(request, "profile.checkNVM.success", Nil, "Profile.checkNVM.success", Json.obj("status" -> "denied", "conditions" -> Json.obj("hasAddress" -> false, "hasPrimaryCrew" -> false, "isActive" -> false))).getResult)
-      //case Some(user) => userService.checkNVM(user).map(_ match {
-	//case Some(nvmState) => WebAppResult.Ok(request, "profile.requestNVM.success", Nil, "Profile.requestNVM.success", Json.obj(nvmState)).getResult
-	//case None => WebAppResult.NotFound(request, "profile.requestNVM.notFound", Nil, "Profile.requestNVM.notFound", Map[String, String]()).getResult
-      //})
+      //case Some(user) => Future.successful(WebAppResult.Ok(request, "profile.checkNVM.success", Nil, "Profile.checkNVM.success", Json.obj("status" -> "denied", "conditions" -> Json.obj("hasAddress" -> false, "hasPrimaryCrew" -> false, "isActive" -> false))).getResult)
+      case Some(user) => {
+        user.profiles.headOption match {
+          case Some(profile) => userService.checkNVM(profile).map(nvmState =>
+            WebAppResult.Ok(request, "profile.checkNVM.success", Nil, "Profile.checkNVM.success", nvmState).getResult
```

Comment:
Same here

Reply:
UserService returns a json. ATM I don't exaclty know how I could change this